### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by hsilva664**
# What does this PR do?

- Added postprocessing test
- Added test to check image from the HUB
- Removed pool attention module, as it is used only in training
- Fixed convert script to accomodate the change in pool latent

### CODE REVIEW

File: src/transformers/models/janus/modular_janus.py [hunk 1]
```diff
@@ -1280,6 +1278,7 @@ def generate(
         if generation_mode == "text":
             # Set to prevent running UnbatchedCFG processor.
             generation_config.guidance_scale = None
+            self.generation_config.guidance_scale = None
```
Comment (hsilva664) [inline src/transformers/models/janus/modular_janus.py]: Added this until we get further guidance on how to avoid the issue. The tests pass with this, although it is hacky.

Comment (yaswanth19) [reply src/transformers/models/janus/modular_janus.py]: @hsilva664 Yes, the only issue with this is if the user want image generation, then he has to again load the model as now guidance scale is None 😓

File: src/transformers/models/janus/modular_janus.py [hunk 2]
```diff
@@ -1396,7 +1397,7 @@ def generate(
 
         for i in range(num_image_tokens):
             model_inputs = self.prepare_inputs_for_generation(
-                inputs_embeds=inputs_embeds, input_ids=input_tokens, attention_mask=attention_mask, **model_kwargs
+                inputs_embeds=inputs_embeds, input_ids=input_tokens, **model_kwargs
```
Comment (yaswanth19) [inline src/transformers/models/janus/modular_janus.py:1399]: I haven't checked whether attention mask is getting updated inplace since you said that it is working fine 😅

File: src/transformers/models/janus/modular_janus.py [hunk 3]
```diff
@@ -1349,6 +1347,9 @@ def generate(
             **model_kwargs,
         )
 
+        # Should only get shape after self._expand_inputs_for_generation
+        batch_size, seq_len = input_ids.shape
```
Comment (yaswanth19) [inline src/transformers/models/janus/modular_janus.py:1350]: Nice catch here 😉

File: tests/models/janus/test_image_processing_janus.py [hunk 4]
```diff
@@ -201,3 +201,28 @@ def test_call_numpy_4_channels(self):
...
+        # Test normalize -> unnormalize back if the arrays match
+        image_inputs = self.image_processor_tester.prepare_image_inputs(equal_resolution=True, torchify=True)
+
+        # Test not batched input
+        encoded_images = image_processing(image_inputs[0], do_resize=False, do_center_crop=False, return_tensors="pt")
+        unnormalized_image = image_processing.postprocess(
+            encoded_images.pixel_values, return_tensors="pt"
+        ).pixel_values
+        # the diff of 1 because the images are in range 0-255 in `uint8` and some precision errors might apply
+        self.assertTrue((image_inputs[0] - unnormalized_image).abs().max() <= 1)
```
Comment (yaswanth19) [inline tests/models/janus/test_image_processing_janus.py]: I think we should refactor it a bit and utilize processor.postprocess (that's what user will utilize) instead of directly utilizing image processor and obviously shift the test to test_processing.

You  can checkout Emu3 testcase which seems to simpler implementation: https://github.com/huggingface/transformers/blob/706703bba6c920b10aa7e7ee8163b06a8a03c450/tests/models/emu3/test_processor_emu3.py#L77

**Review by yaswanth19 — APPROVED** [review]
Left a single comment for testing postprocess otherwise GTM

File: tests/models/janus/test_image_processing_janus.py (hunk 4 above)
Comment (hsilva664) [reply tests/models/janus/test_image_processing_janus.py]: Will look. The current one was adapted from Anole, so I thought it would be fine.
